### PR TITLE
Feature/cleanup replication sync tailing metric statistics

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

Added/modified metrics:
```
> arangodb_replication_synchronous_requests_total_number_total (Heiko)		
> arangodb_replication_synchronous_requests_total_time_total (Heiko) 		
> arangodb_replication_tailing_apply_time_total (Heiko)					
> arangodb_replication_tailing_bytes_received_total (Heiko) 
> arangodb_replication_tailing_documents_total (Heiko)
> arangodb_replication_tailing_follow_tick_failures_total (Heiko)
> arangodb_replication_tailing_markers_total (Heiko)
> arangodb_replication_tailing_removals_total (Heiko)
> arangodb_replication_tailing_requests_total (Heiko)
> arangodb_replication_tailing_request_time_total (Heiko)
```

- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: Documentation/Metrics/arangodb_replication_synchronous_requests_total_time_total.yaml [hunk 1]
```diff
@@ -9,4 +9,5 @@ complexity: medium
 exposedBy:
   - dbserver
 description: |
-  Total time needed for all synchronous replication requests.
+  The total time needed for all synchronous replication requests
+  between DB-Servers being done.
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_synchronous_requests_total_time_total.yaml:13]:
```suggestion
  between dbservers being done.
```

File: Documentation/Metrics/arangodb_replication_tailing_apply_time_total.yaml [hunk 2]
```diff
@@ -8,5 +8,13 @@ category: Replication
...
-  Accumulated time needed to apply replication tailing data.
+description:
+  The accumulated time needed to locally process the continuous
+  replication log on a Follower received from a replication
+  Leader.
+throubleshoot:
+  If you see unusual spikes here, the follower might not have enough
+  IO bandwith or might be overloaded. Try to provision more IOPS or
+  more CPU capacity. Additionally, it could make sense to compare the
+  value with all other available Follower DB-Servers to detect potential
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_apply_time_total.yaml:19]:
```suggestion
  value with all other available follower dbservers to detect potential
```

File: Documentation/Metrics/arangodb_replication_tailing_apply_time_total.yaml [hunk 3]
```diff
@@ -8,5 +8,13 @@ category: Replication
 complexity: medium
 exposedBy:
   - dbserver
-description: |
-  Accumulated time needed to apply replication tailing data.
+description:
+  The accumulated time needed to locally process the continuous
+  replication log on a Follower received from a replication
+  Leader.
+throubleshoot:
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_apply_time_total.yaml:15]:
```suggestion
throubleshoot: |
```

File: Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml [hunk 4]
```diff
@@ -8,5 +8,12 @@ category: Replication
 complexity: medium
 exposedBy:
   - dbserver
-description: |
-  Accumulated number of bytes received for replication tailing requests.
+description:
+  The accumulated number of bytes received from a Leader for replication
+  tailing requests. The higher the amount of bytes is, the more data
+  is being processed afterwards on the Follower DB-Server.
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml:14]:
```suggestion
  is being processed afterwards on the follower dbserver.
```

File: Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml [hunk 5]
```diff
@@ -8,5 +8,12 @@ category: Replication
...
+troubleshoot:
+  Compare this metric with all other related participating Follower
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml:16]:
```suggestion
  Compare this metric with all other related participating follower
```

File: Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml [hunk 6]
```diff
@@ -8,5 +8,12 @@ category: Replication
...
+  DB-Servers. If the given value on a DB-Server is considerable higher,
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml:17]:
```suggestion
  dbservers. If the given value on a dbserver is considerable higher,
```

File: Documentation/Metrics/arangodb_replication_tailing_documents_total.yaml [hunk 7]
```diff
@@ -8,5 +8,16 @@ category: Replication
 complexity: medium
 exposedBy:
   - dbserver
-description: |
-  Accumulated number of replication tailing document inserts/replaces processed.
+description:
+  The accumulated number of replication tailing document inserts/replaces
+  processed on a Follower.
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_documents_total.yaml:13]:
```suggestion
  processed on a follower.
```

File: Documentation/Metrics/arangodb_replication_tailing_documents_total.yaml [hunk 8]
```diff
@@ -8,5 +8,16 @@ category: Replication
...
+troubleshoot:
+  Compare this metric with all other related participating Follower
+  DB-Servers. If the given value on a DB-Server is considerable higher,
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_documents_total.yaml:16]:
```suggestion
  Compare this metric with all other related participating follower
  dbservers. If the given value on a dbserver is considerable higher,
```

File: Documentation/Metrics/arangodb_replication_tailing_documents_total.yaml [hunk 9]
```diff
@@ -8,5 +8,16 @@ category: Replication
...
+  you might want to think about rebalancing your data as the overall
+  work might not be evenly distributed. It is important to understand
+  that this metric only enumerates the amount of documents and does not
+  compare document sizes. Even if values compared to other DB-Servers
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_documents_total.yaml:20]:
```suggestion
  compare document sizes. Even if values compared to other dbservers
```

File: Documentation/Metrics/arangodb_replication_tailing_follow_tick_failures_total.yaml [hunk 10]
```diff
@@ -9,4 +9,10 @@ complexity: advanced
 exposedBy:
   - dbserver
 description: |
-  Number of replication tailing failures due to missing tick on leader.
+  The number of replication tailing failures due to missing tick on leader.
+troubleshoot:
+  Action is required. Required follower ticks is not present (potentially
+  removed) on a Leader DB-Server. Please check the related Leader DB-Server
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_follow_tick_failures_total.yaml:15]:
```suggestion
  If this is non-zero, action is required. A required follower tick is not present (potentially
  removed) on a leader dbserver. Please check the related leader dbserver
```

File: Documentation/Metrics/arangodb_replication_tailing_markers_total.yaml [hunk 11]
```diff
@@ -9,4 +9,9 @@ complexity: medium
 exposedBy:
   - dbserver
 description: |
-  Number of replication tailing markers processed.
+  The number of replication tailing markers processed on a Follower
+  DB-Server. Markers are specific operations which are part of the
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_markers_total.yaml:13]:
```suggestion
  The number of replication tailing markers processed on a follower
  dbserver. Markers are specific operations which are part of the
```

File: Documentation/Metrics/arangodb_replication_tailing_removals_total.yaml [hunk 12]
```diff
@@ -9,4 +9,5 @@ complexity: medium
 exposedBy:
   - dbserver
 description: |
-  Number of replication tailing document removals processed.
+  The amount of document removal based marker operations on a
+  Follower DB-Server.
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_removals_total.yaml:13]:
```suggestion
  The amount of document removal based marker operations on a
  follower dbserver.
```

File: Documentation/Metrics/arangodb_replication_tailing_request_time_total.yaml [hunk 13]
```diff
@@ -10,3 +10,10 @@ exposedBy:
   - dbserver
 description: |
   Aggregated wait time for replication tailing requests.
+troubleshoot:
+  If you see unusual spikes here, please insepect potential
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_request_time_total.yaml:14]:
```suggestion
  If you see unusual spikes here, please inspect potential
```

File: Documentation/Metrics/arangodb_replication_tailing_request_time_total.yaml [hunk 14]
```diff
@@ -10,3 +10,10 @@ exposedBy:
...
+  network issues. It may help to increase network bandwith
+  and/or reduce network latency. In case there are no network
+  issues, also check the load of the serving Leader DB-Server,
+  as well as the Follower DB-Server, as they could potentially
```
Comment (neunhoef) [inline Documentation/Metrics/arangodb_replication_tailing_request_time_total.yaml:18]:
```suggestion
  issues, also check the load of the serving leader dbserver,
  as well as the follower dbserver, as they could potentially
```

**Review by neunhoef — CHANGES_REQUESTED** [review]
Only small typo suggestions. Otherwise: LGTM.

File: Documentation/Metrics/arangodb_replication_tailing_apply_time_total.yaml [hunk 15]
```diff
@@ -9,4 +9,12 @@ complexity: medium
 exposedBy:
   - dbserver
 description: |
-  Accumulated time needed to apply replication tailing data.
+  The accumulated time needed to locally process the continuous
+  replication log on a follower received from a replication
+  Leader.
```
Comment (hkernbach) [inline Documentation/Metrics/arangodb_replication_tailing_apply_time_total.yaml:14]:
```suggestion
  leader.
```

File: Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml [hunk 16]
```diff
@@ -8,5 +8,12 @@ category: Replication
 complexity: medium
 exposedBy:
   - dbserver
-description: |
-  Accumulated number of bytes received for replication tailing requests.
+description:
+  The accumulated number of bytes received from a Leader for replication
```
Comment (hkernbach) [inline Documentation/Metrics/arangodb_replication_tailing_bytes_received_total.yaml:12]:
```suggestion
  The accumulated number of bytes received from a leader for replication
```

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**